politeiawww/backend.go

```diff
@@ -1949,12 +1949,30 @@ func (b *backend) ProcessProposalDetails(propDetails www.ProposalsDetails, user
 	cachedProposal := b._convertPropFromInventoryRecord(p)
 	b.RUnlock()
 
+	propIsPublic := cachedProposal.Status == www.PropStatusPublic || cachedProposal.Status == www.PropStatusAbandoned
+
+	// validate requested version when it is specified and the proposal is public
+	if propDetails.Version != "" && propIsPublic {
```

Comment:
Do you need to check if the proposal is public?  The `lastestVersion` for a non-public proposal is 1, so any attempt to request a version > 1 is going to throw the `ErrorStatusInvalidPropVersion` anyway.